module/os/freebsd/zfs/abd_os.c

```diff
@@ -0,0 +1,460 @@
+/*
+ * This file and its contents are supplied under the terms of the
+ * Common Development and Distribution License ("CDDL"), version 1.0.
+ * You may only use this file in accordance with the terms of version
+ * 1.0 of the CDDL.
+ *
+ * A full copy of the text of the CDDL should have accompanied this
+ * source.  A copy of the CDDL is also available via the Internet at
+ * http://www.illumos.org/license/CDDL.
+ */
+
+/*
+ * Copyright (c) 2014 by Chunwei Chen. All rights reserved.
+ * Copyright (c) 2016 by Delphix. All rights reserved.
+ */
+
+/*
+ * ARC buffer data (ABD).
+ *
```

Comment:
It looks like this diagram is the same in abd.c and both abd_os.c's.  Let's remove it from the abd_os.c's, and perhaps add a comment to those files indicating that these are implementing the OS-specific parts of ABD, to see abd.c for an overview, and then talk about OS-specific considerations (e.g. the HIGHMEM stuff for linux).

Reply:
Sounds good. I will update this.